sync3/handler/ensure_polling.go

```diff
@@ -93,6 +104,21 @@ func (p *EnsurePoller) OnInitialSyncComplete(payload *pubsub.V2InitialSyncComple
 	close(ch)
 }
 
+func (p *EnsurePoller) OnExpiredToken(payload *pubsub.V2ExpiredToken) {
+	pid := sync2.PollerID{UserID: payload.UserID, DeviceID: payload.DeviceID}
+	p.mu.Lock()
+	defer p.mu.Unlock()
+	pending, exists := p.pendingPolls[pid]
+	if !exists {
+		// We weren't tracking the state of this poller, so we have nothing to clean up.
+		return
+	}
+	if pending.ch != nil {
+		close(pending.ch)
```

Comment:
```suggestion
		// unblock anything waiting on this sync, it's never going to happen now.
		close(pending.ch)
```

Reply:
Missed this, will include in a future PR